### PR TITLE
Implement event hub reactive operation

### PR DESCRIPTION
**Pull request by warrenzhu25**

### CODE REVIEW

File: spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/eventhub/AbstractEventHubTemplate.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    private CheckpointMode checkpointMode = CheckpointMode.BATCH;
+
+    AbstractEventHubTemplate(EventHubClientFactory clientFactory) {
+        this.clientFactory = clientFactory;
+    }
+
+    private static EventProcessorOptions buildEventProcessorOptions(StartPosition startPosition) {
+        EventProcessorOptions options = EventProcessorOptions.getDefaultOptions();
+
+        if (startPosition == StartPosition.EARLISET) {
```
Comment (Incarnation-p-lee) [inline spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/eventhub/AbstractEventHubTemplate.java]: suggest to use enum with `Function` here.

File: spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/eventhub/AbstractEventHubTemplate.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    protected void register(Tuple<String, String> nameAndGroup, EventHubProcessor eventProcessor) {
+        EventProcessorHost host = this.clientFactory.getProcessorHostCreator().apply(nameAndGroup);
+        host.registerEventProcessorFactory(context -> eventProcessor, buildEventProcessorOptions(startPosition));
+    }
+
+    protected void unregister(Tuple<String, String> nameAndGroup) {
+        this.clientFactory.getProcessorHostCreator().apply(nameAndGroup).unregisterEventProcessor()
+                          .whenComplete((s, t) -> {
+                              if (t != null) {
```
Comment (Incarnation-p-lee) [inline spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/eventhub/AbstractEventHubTemplate.java]: seems this piece of code pattern appears more than one place, wonder if we can add one `Function` for this.

File: spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/eventhub/AbstractEventHubTemplate.java (hunk 1 above)
Comment (warrenzhu25) [reply spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/eventhub/AbstractEventHubTemplate.java]: what do you mean by this?

File: spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/eventhub/AbstractEventHubTemplate.java (hunk 2 above)
Comment (warrenzhu25) [reply spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/eventhub/AbstractEventHubTemplate.java]: only log might not need another function

**Pull request merged**